### PR TITLE
Sharding IO tests(amazon web services and amazon web services 2) from java post commit task

### PR DESCRIPTION
**Pull request by andoni-guzman**
Sharding IO tests Amazon web services and Amazon web services 2 from Java PostCommit task
Looking to reduce JavaPostCommit task some test were split in different ones. This PR is for Amazon web services and Amazon web services 2 tests. 
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .test-infra/jenkins/job_PreCommit_Java_AWS_IO_Direct.groovy [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+PrecommitJobBuilder builder = new PrecommitJobBuilder(
+  scope: this,
+  nameBase: 'Java_Amazon-Web-Services_IO_Direct',
+  gradleTask: ':sdks:java:io:amazon-web-services:integrationTest',
+  gradleSwitches: [
+    '-PdisableSpotlessCheck=true'
+  ], // spotless checked in separate pre-commit
+  triggerPathPatterns: [
+    '^sdks/java/io/amazon-web-services/.*$',
+    '^runners/direct-java/.*$',
```
Comment (kennknowles) [inline .test-infra/jenkins/job_PreCommit_Java_AWS_IO_Direct.groovy:30]: Remove this trigger path too.

File: .test-infra/jenkins/job_PreCommit_Java_AWS2_IO_Direct.groovy [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+PrecommitJobBuilder builder = new PrecommitJobBuilder(
+  scope: this,
+  nameBase: 'Java_Amazon-Web-Services2_IO_Direct',
+  gradleTask: 'sdks:java:io:amazon-web-services2:integrationTest',
+  gradleSwitches: [
+    '-PdisableSpotlessCheck=true'
+  ], // spotless checked in separate pre-commit
+  triggerPathPatterns: [
+    '^sdks/java/io/amazon-web-services2/.*$',
+    '^runners/direct-java/.*$',
```
Comment (kennknowles) [inline .test-infra/jenkins/job_PreCommit_Java_AWS2_IO_Direct.groovy:30]: We do not need to run these tests when the direct runner is modified, since these tests are not for checking the correctness of the direct runner. So should remove this trigger path.

**Review by kennknowles — CHANGES_REQUESTED** [review]

**Comment by andoni-guzman** [comment]
Hi @kennknowles  
I delete the trigger for direct runner. Please let me know if you want to separate the task and in how way you want it or if it's okay in this way.
Thank you!

File: build.gradle.kts [hunk 3]
```diff
@@ -197,8 +197,7 @@ tasks.register("javaPostCommit") {
   dependsOn(":sdks:java:extensions:google-cloud-platform-core:postCommit")
   dependsOn(":sdks:java:extensions:zetasketch:postCommit")
   dependsOn(":sdks:java:io:google-cloud-platform:postCommit")
-  dependsOn(":sdks:java:io:amazon-web-services:integrationTest")
-  dependsOn(":sdks:java:io:amazon-web-services2:integrationTest")
+  dependsOn(":sdks:java:io:kinesis:integrationTest")
```
Comment (kennknowles) [inline build.gradle.kts:200]: Why is this added?

Comment (andoni-guzman) [reply build.gradle.kts:200]: I think is related to a merge conflict with other branch that is working in the tests. Let me talk about it with the team and do the changes.

**Comment by kennknowles** [comment]
I suggest doing a `git rebase origin/master` and then force pushing to the PR.

**Comment by mosche** [comment]
@andoni-guzman wondering, should this be using the PostcommitJobBuilder, or is it intentional to have it as a precommit job from now on?

**Comment by andoni-guzman** [comment]
Hi @kennknowles,
I think it's ready for review, let me know if something it's missing c:

**Review by kennknowles — APPROVED** [review]
LGTM but needs yet another rebase since there is a conflict with master
